Sources/StreamChat/Controllers/ChannelController/ChannelController.swift

```diff
@@ -791,20 +820,95 @@ public extension ChatChannelController {
+        }
 
+        isLoadingNextMessages = true
+        channelQuery.pagination = MessagesPagination(pageSize: limit, parameter: .greaterThan(messageId))
         updater.update(channelQuery: channelQuery, isInRecoveryMode: false, completion: { result in
-            self.callback { completion?(result.error) }
+            self.isLoadingNextMessages = false
+            switch result {
+            case let .success(payload):
+                self.updateNewestFetchedMessageId(with: payload)
+                self.callback { completion?(nil) }
+            case let .failure(error):
+                self.callback { completion?(error) }
+            }
         })
     }
 
+    /// Load messages around the given message id. Useful to jump to a message which is not loaded yet.
+    ///
+    /// Cleans the current messages of the channel and loads the message with the given id,
+    /// and the messages around it depending on the limit provided.
+    ///
+    /// Ex: If the limit is 25, it will load the message and 12 on top and 12 below it. (25 total)
+    ///
+    /// - Parameters:
+    ///   - messageId: The message id of the message to jump to.
+    ///   - limit: The number of messages to load in total, including the message to jump to.
+    ///   - completion: Callback when the API call is completed.
+    func loadPageAroundMessageId(_ messageId: MessageId, limit: Int? = nil, completion: ((Error?) -> Void)? = nil) {
+        guard let cid = self.cid, isChannelAlreadyCreated else {
+            channelModificationFailed(completion)
+            return
+        }
+
+        guard !isJumpingToMessage else {
+            return
+        }
+
+        isJumpingToMessage = true
+
+        let limit = limit ?? channelQuery.pagination?.pageSize ?? .messagesPageSize
+        channelQuery.pagination = MessagesPagination(pageSize: limit, parameter: .around(messageId))
+        updater.update(
+            channelQuery: channelQuery,
+            isInRecoveryMode: false,
+            onBeforeSavingChannel: { session in
+                session.deleteChannelMessages(cid: cid)
+            },
+            completion: { result in
+                self.isJumpingToMessage = false
+
+                switch result {
+                case let .success(payload):
+                    self.updateNewestFetchedMessageId(with: payload)
+                    self.updateOldestFetchedMessageId(with: payload)
+                    self.callback { completion?(nil) }
+                case let .failure(error):
+                    log.error("Not able to load message around messageId: \(messageId). Error: \(error)")
+                    self.callback { completion?(error) }
+                }
+            }
+        )
+    }
+
+    /// Cleans the current state and loads the first page again.
+    /// - Parameter completion: Callback when the API call is completed.
+    func loadFirstPage(_ completion: ((_ error: Error?) -> Void)? = nil) {
+        isJumpingToMessage = true
```

Comment:
Not, in this case, no, because you might be jumping to the middle of a page, and then while jumping press the scroll to the bottom page, and you want basically override that previous action.